client/blocks/jitm/templates/sidebar-banner.jsx

```diff
@@ -1,15 +1,57 @@
 /**
- * External Dependencies
+ * External dependencies
  */
 import React from 'react';
 import Gridicon from 'components/gridicon';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import UpsellNudge from 'blocks/upsell-nudge';
+import { abtest } from 'lib/abtest';
 
 /**
  * Style dependencies
  */
 import './sidebar-banner.scss';
 
-export default function SidebarBannerTemplate( { icon, CTA, message, onClick, trackImpression } ) {
+export default function SidebarBannerTemplate( {
+	icon,
+	CTA,
+	message,
+	onClick,
+	trackImpression,
+	id,
+	event,
+	featureClass,
+	tracks,
+	...props
+} ) {
+	if ( abtest( 'sidebarUpsellNudgeUnification' ) === 'variantShowUnifiedUpsells' ) {
+		const clickName = get( tracks, [ 'click', 'name' ] ) || `jitm_nudge_click_${ id }`;
+		const clickProps = get( tracks, [ 'click', 'props' ] );
+		const displayName = get( tracks, [ 'display', 'name' ] ) || `jitm_nudge_impression_${ id }`;
+		const displayProps = get( tracks, [ 'display', 'props' ] );
+		const jitmProps = { id: id, jitm: true };
+
+		return (
+			<UpsellNudge
+				callToAction={ CTA.message }
+				compact
+				event={ displayName }
+				href={ CTA.link }
+				onClick={ onClick }
+				title={ message }
+				tracksClickName={ clickName }
+				tracksClickProperties={ { ...jitmProps, ...clickProps } }
+				tracksImpressionName={ displayName }
+				tracksImpressionProperties={ { ...jitmProps, ...displayProps } }
+				{ ...props }
```

Comment:
What other props are we interested in passing? since this is a pretty specialized component instead of say a plain html element I think it's better to be explicit here.

Reply:
[`onDismiss`](https://github.com/Automattic/wp-calypso/blob/1c7e8f8e8c5563a9c89603c586c5b3bfd7ac6478/client/blocks/jitm/index.jsx#L50-L54) could be one of them, but I think the new `UpsellNudge` component expects a `onDismissClick` property. That will also require to pass down the `tracksDismiss*` properties. 

Although seems that none of the JITMs in the sidebar can be dismissed.

Reply:
Yes, Miguel is right, I was thinking the dismiss properties; they're only used by the "Add Another Domain" nudge right now. But you're right, it's better to be explicit.